app/upgrades/v11/upgrades.go

```diff
@@ -38,7 +47,42 @@ func CreateUpgradeHandler(
 
 		MigrateEscrowAccounts(ctx, ak)
 
-		// Leave modules are as-is to avoid running InitGenesis.
+		// create ICS27 Controller submodule params, with the controller module NOT enabled
+
+		gs := &icatypes.GenesisState{
+			ControllerGenesisState: icatypes.ControllerGenesisState{},
+			HostGenesisState: icatypes.HostGenesisState{
+				Port: icatypes.PortID,
+				Params: icahosttypes.Params{
+					HostEnabled: true,
+					AllowMessages: []string{
+						sdk.MsgTypeURL(&banktypes.MsgSend{}),
+						sdk.MsgTypeURL(&banktypes.MsgMultiSend{}),
+						sdk.MsgTypeURL(&distrtypes.MsgSetWithdrawAddress{}),
+						sdk.MsgTypeURL(&distrtypes.MsgWithdrawDelegatorReward{}),
+						sdk.MsgTypeURL(&govtypes.MsgVote{}),
+						sdk.MsgTypeURL(&govtypes.MsgVoteWeighted{}),
+						sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
+						sdk.MsgTypeURL(&stakingtypes.MsgUndelegate{}),
+						sdk.MsgTypeURL(&stakingtypes.MsgCancelUnbondingDelegation{}),
+						sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
+						sdk.MsgTypeURL(&transfertypes.MsgTransfer{}),
+					},
+				},
+			},
+		}
+
+		// Register the consensus version in the version map to avoid the SDK from triggering the default
+		// InitGenesis function.
+		vm[icatypes.ModuleName] = ica.AppModule{}.ConsensusVersion()
+
+		bz, err := icatypes.ModuleCdc.MarshalJSON(gs)
+		if err != nil {
+			return nil, errorsmod.Wrapf(err, "failed to marshal %s genesis state", icatypes.ModuleName)
+		}
```

Comment:
nit regarding code structure: do marshalling of genesis state directly after `gs` definition
```suggestion
		bz, err := icatypes.ModuleCdc.MarshalJSON(gs)
		if err != nil {
			return nil, errorsmod.Wrapf(err, "failed to marshal %s genesis state", icatypes.ModuleName)
		}
		
		// Register the consensus version in the version map to avoid the SDK from triggering the default
		// InitGenesis function.
		vm[icatypes.ModuleName] = ica.AppModule{}.ConsensusVersion()
```